### PR TITLE
BACK-28: Implement endpoint serving mock data

### PR DESCRIPTION
**Pull request by thomas-woehrle**
## Description of Issue
Frontend should be able to receive mock repositories in the format specified in the notion file. 
_I know that the term "endpoint" in the title is not exactly correct when working with GraphQL._
## How I implemented
I adapted the schema requested by the frontend and added it to the schema. 
Then I build resolvers. 
I let chatGPT create sample projects and added them to a separate file. 
## Other
1. There are 9 sample projects which are duplicated twice. This way 27 projects are returned which simulates the typical amount.
2. I used git commit --no-verify because of problems with the pre-commit-hook. I am serving mock data from a json file and do not want to duplicate all GraphQL types as Typescript types. Therefore no type is specified in the resolver -> pre-commit hook complains
3. I get an error because of importing the json file. Everything works still and I could not find a fix for this.

### CODE REVIEW

**Comment by pczern** [comment]
very good, please let me know how we will link the Hackernews, and GitHub Repo

File: src/graphql/schema.ts [hunk 1]
```diff
@@ -1,15 +1,111 @@
...
+  }
+
+  type DiscordChannel implements SocialMediaPresence {
+    platform: String!
+    accountName: String!
+    link: String!
+    memberCount: Int!
+  }
+
+  interface SocialMediaTopPost {
```
Comment (pczern) [inline src/graphql/schema.ts:72]: can we do like `SocialMediaPost`? And then use a filter to get the top ones?

**Review by pczern — APPROVED** [review]
Please have a look at the comment, we should definitely move to Graphql filter for top posts long term

**Comment by thomas-woehrle** [comment]
Will not merge for now. Have to address the eslint problem (-> possibly add GraphQL codegen) and maybe integrate the enums requested by frontend.

**Pull request merged**